src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs

```diff
@@ -525,15 +525,22 @@ public void OnDemandPayoutAsync()
 
                 if(b.BaseFeePerGas <= 0) return;
 
-                if(b.BaseFeePerGas <= extraConfig.MaxGasLimit)
+                if(b.BaseFeePerGas <= (extraConfig.MaxGasLimit * extraConfig.TopMinersGasLimit))
                 {
                     if(ondemandPayTask == null || ondemandPayTask.IsCompleted)
                     {
-                        ondemandPayCts?.Dispose();
-                        ondemandPayCts = new CancellationTokenSource();
+                        for(float x = 1; x <= extraConfig.TopMinersGasLimit; x += extraConfig.TopMinersGasLimitFactor)
```

Comment:
So this has us paying out at 3 discrete gas fees,  1x MaxGasLimit, 2x MaxGasLimit and 3x MaxGasLimit, right?

I think what Matt had in mind was a dynamic ratio.  like this:

```
newMaxGasFee = 3 x MaxGasLimit

if(currentGasFee < newMaxGasFee)
	PayoutBalancesOverThresholdAsync
	
for each balance
	feeFactor = payoutThreshold / minerBalance
	if(currentGasFee <= MaxGasLimit * feeFactor)
		payout this balance
	else
		gas still higher than the factor.  don't pay this guy.
```